homeassistant/components/modbus/__init__.py

```diff
@@ -216,6 +225,15 @@
             vol.Optional(CONF_MIN_TEMP, default=5): cv.positive_int,
             vol.Optional(CONF_STEP, default=0.5): vol.Coerce(float),
             vol.Optional(CONF_TEMPERATURE_UNIT, default=DEFAULT_TEMP_UNIT): cv.string,
+            vol.Optional(CONF_HVAC_MODE_REGISTER): cv.positive_int,
```

Comment:
It is better to have CONF_HVAC_MODE as optional, and the below it the different modes.

Reply:
This was my initial thought also, but I tried to follow the convention used in MQTT configuration, where both the name of the register (topic in MQTT) and the set of possible values are defined at the same level. If you insist, I can change it but I think it might confuse users. This is also the reason I needed extra constant for each possible mode.

Reply:
Yes I insist, because that is what Modbus does with other parameters of similar type (e.g. 'verify').